src/metadata/MetadataStorage.ts

```diff
@@ -35,14 +35,26 @@ export class MetadataStorage {
    * Adds a new validation metadata.
    */
   addValidationMetadata(metadata: ValidationMetadata): void {
-    this.validationMetadatas.push(metadata);
+    const existingMetadata = this.validationMetadatas.get(metadata.target);
+    if (existingMetadata) {
+      existingMetadata.push(metadata);
+      this.validationMetadatas.set(metadata.target, existingMetadata);
```

Comment:
This looks unnecessary, the key->value combination already exists as push does not change the array identity.

Reply:
This is fixed in the new PR.